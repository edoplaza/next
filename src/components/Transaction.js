import React, { Component } from 'react';
import moment from 'moment';

class Transaction extends Component{
  render(){
    let from = this.props.from
    let to = this.props.to
    let description = this.props.description
    let amount = this.props.amount
    let date = this.props.date
    return (
      <li className="transaction">
         <table className="transaction__table">
           <tbody>
              { from &&
               <tr>
                 <th>From:</th>
                 <td>{from}</td>
               </tr>
              }
              { to &&
                <tr>
                  <th>To:</th>
                  <td>{to}</td>
                </tr>
              }
              <tr>
               <th>description:</th>
               <td>{description}</td>
              </tr>
              <tr>
               <th>amount:</th>
               <td>{amount}<span> â‚¬</span></td>
              </tr>
              <tr>
               <th>Date:</th>
              <td>{ moment(date).format("MMM DD, YYYY") }</td>
              </tr>
           </tbody>
         </table>
       </li>
    );
  }
}

export default Transaction
