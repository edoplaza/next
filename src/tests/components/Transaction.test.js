import React from 'react';
import { shallow } from 'enzyme';
import Transaction from '../../components/Transaction';

test('should render Transaction correctly', () => {
  const wrapper = shallow(<Transaction />);
  expect(wrapper).toMatchSnapshot();
})


