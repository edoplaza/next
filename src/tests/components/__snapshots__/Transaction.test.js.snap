// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Transaction correctly 1`] = `
<li
  className="transaction"
>
  <table
    className="transaction__table"
  >
    <tbody>
      <tr>
        <th>
          description:
        </th>
        <td />
      </tr>
      <tr>
        <th>
          amount:
        </th>
        <td>
          <span>
             â‚¬
          </span>
        </td>
      </tr>
      <tr>
        <th>
          Date:
        </th>
        <td>
          May 18, 2018
        </td>
      </tr>
    </tbody>
  </table>
</li>
`;
